mirrors_countme/totals.py

```diff
@@ -118,38 +118,110 @@ class RawDB(SQLiteReader):
     def __init__(self, filename, **kwargs):
         super().__init__(filename, CountmeItem, tablename="countme_raw", **kwargs)
 
+    @property
+    def mintime(self):
+        return self.mintime_countme
+
+    @property
+    def maxtime(self):
+        return self.maxtime_countme
+
     def complete_weeks(self):
         """Return a range(startweek, provweek) that covers (valid + complete)
         weeknums contained in this database. The database may contain some
         data for `provweek`, but since it's provisional/incomplete it's
         outside the range."""
+        if self.mintime is None:
+            return []
         # startweek can't be earlier than the first week of data
         startweek = max(weeknum(self.mintime), COUNTME_START_WEEKNUM)
         # A week is provisional until the LOG_JITTER_WINDOW expires, so once
         # tsmax minus LOG_JITTER_WINDOW ticks over into a new weeknum, that
         # weeknum is the provisional one. So...
-        provweek = weeknum(self.maxtime - LOG_JITTER_WINDOW)
+        provweek = max(weeknum(self.maxtime - LOG_JITTER_WINDOW), COUNTME_START_WEEKNUM)
         return range(startweek, provweek)
 
-    def week_count(self, weeknum):
+    def week_iter(self, weeknum, select: tuple | list):
+        item_select = ",".join(select)
         start_ts = weeknum * WEEK_LEN + COUNTME_EPOCH
         end_ts = start_ts + WEEK_LEN
-        cursor = self._connection.execute(
-            f"SELECT COUNT(*)"
+        return self._connection.execute(
+            f"SELECT {item_select}"
             f" FROM {self._tablename}"
-            f" WHERE timestamp >= {start_ts} AND timestamp < {end_ts}"
+            f" WHERE timestamp >= {start_ts} AND timestamp < {end_ts} AND sys_age >= 0"
         )
+
+    def week_count(self, weeknum):
+        cursor = self.week_iter(weeknum, ("COUNT(*)",))
         return cursor.fetchone()[0]
 
+
+class RawDBU(RawDB):
+    def __init__(self, fp, **kwargs):
+        super().__init__(fp, **kwargs)
+
+    @property
+    def mintime(self):
+        return self.mintime_unique
+
+    @property
+    def maxtime(self):
+        return self.maxtime_unique
```

Comment:
Depending on if we want to count `countme`, this probably should use `mintime`, `maxtime` properties which cover all log entries (i.e. probably requires that we have a base class which doesn’t override `mintime`, `maxtime` like `RawDB` does above).